src/main/java/io/supertokens/usermetadata/UserMetadata.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *    Copyright (c) 2020, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.usermetadata;
+
+import javax.annotation.Nonnull;
+
+import com.google.gson.JsonObject;
+
+import io.supertokens.Main;
+import io.supertokens.utils.MetadataUtils;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.usermetadata.sqlStorage.UserMetadataSQLStorage;
+import io.supertokens.storageLayer.StorageLayer;
+
+public class UserMetadata {
+    public static JsonObject updateUserMetadata(Main main, @Nonnull String userId, @Nonnull JsonObject metadataUpdate)
+            throws StorageQueryException, StorageTransactionLogicException {
+        UserMetadataSQLStorage storage = StorageLayer.getUserMetadataStorage(main);
+
+        return storage.startTransaction((con) -> {
+            JsonObject originalMetadata = storage.getUserMetadata_Transaction(con, userId);
+
+            JsonObject updatedMetadata = originalMetadata == null ? new JsonObject() : originalMetadata;
+            MetadataUtils.shallowMergeMetadataUpdate(updatedMetadata, metadataUpdate);
+
+            storage.setUserMetadata_Transaction(con, userId, updatedMetadata);
```

Comment:
Add explicit testing for if the select statement returns an empty result and by the time it comes here, another thread has already inserted the row. Then what happens here? Does it retry?

Reply:
Added the test in the Postgres and MySQL PRs.

Reply:
okay